app/controllers/application_controller.rb

```diff
@@ -20,6 +33,12 @@ def configure_permitted_parameters
 
     def user_not_authorized
       flash[:warning] = "You are not authorized to perform this action."
-      redirect_to(request.referrer || root_path)
+      redirect_to(request.referer || root_path)
+    end
+
+    # finds most viewed article created recently
+    def most_viewed
+      Article.where("created_at >= :week_ago AND views = :max_views",
+                    week_ago: DateTime.now.change(day: 7), max_views: Article.maximum("views")).first
```

Comment:
Consider using counter cache in the future. (add to TODO, not in this PR)
https://blog.appsignal.com/2018/06/19/activerecords-counter-cache.html
https://guides.rubyonrails.org/association_basics.html#options-for-belongs-to-counter-cache
https://www.dailysmarty.com/posts/how-to-count-page-impressions-in-rails